src/Illuminate/Cache/FileStore.php

```diff
@@ -111,7 +111,9 @@ public function put($key, $value, $minutes)
     protected function createCacheDirectory($path)
     {
         try {
-            $this->files->makeDirectory(dirname($path), 0777, true, true);
+            if (! file_exists(dirname($path))) {
```

Comment:
Thanks for feedback but I didn't see your main point, honestly. Please correct me if my understanding was wrong. The purpose of putting this check inside createCacheDirectory is not to throw exception when we put cache with the same key. If I put this check on file object ( I assume, Illuminate\Filesystem\Filesystem ), it will effect other features that will call makeDirectory method. I think, it might need this exception in other case.